node/tessel-export.js

```diff
@@ -77,6 +77,24 @@ Tessel.Port = function(name, socketPath, board) {
     }
   });
 
+  // Unreference this socket so that the script will exit
+  // if nothing else is waiting in the event queue.
+  this.sock.unref();
+
+   // Number of tasks occupying the socket
+  this.sock.references = 0;
+
+  this.sock._ref = function() {
+    this.sock.references++;
+    this.sock.ref();
+  }.bind(this);
+
+  this.sock._unref = function() {
+    if (--this.sock.references === 0) {
+      this.sock.unref();
+    }
+  }.bind(this);
```

Comment:
If the point is to `ref` and `unref` the socket on behalf of the `port`, while abiding some `port`-specific state, then just make these: 

```js
Tessel.Port = function(...) {
  // ...
  // Track number of tasks occupying this port's socket
  this.pending = 0;

  // Unreference this port's socket so that the script will exit
  // if nothing else is waiting in the event queue.
  this.unref();

  // ...
}

Tessel.Port.prototype.ref = function() {
  this.pending++;
  this.sock.ref();
};

Tessel.Port.prototype.unref = function() {
  // This will prevent pending from dropping below 0
  // and avoid mixing this.sock.unref() calls
  if (this.pending > 0) {
    this.pending--;
  }
  
  if (this.pending === 0) {
    this.sock.unref();
  }
};
```

- all calls to `this.sock._unref()` become `this.unref()`
- all calls to `this.sock._ref()` become `this.ref()`